cmd/group-auto-updater/main.go

```diff
@@ -0,0 +1,143 @@
+// The purpose of this tool is to read a peribolos configuration
+// file, get the admins/members of a given organization and
+// update the users of a specific group in an Openshift cluster.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/config/org"
+
+	userV1 "github.com/openshift/client-go/user/clientset/versioned/typed/user/v1"
+
+	"github.com/openshift/ci-tools/pkg/steps"
+	"github.com/openshift/ci-tools/pkg/util"
+)
+
+type options struct {
+	group           string
+	peribolosConfig string
+	org             string
+
+	dryRun            bool
+	determinizeOutput bool
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.StringVar(&o.group, "group", "", "The group that will be updated in the cluster.")
+	fs.StringVar(&o.peribolosConfig, "peribolos-config", "", "Peribolos configuration file")
+	fs.StringVar(&o.org, "org", "", "Org from peribolos configuration")
+
+	fs.BoolVar(&o.dryRun, "dry-run", false, "Print the generated group without updating it")
+	fs.BoolVar(&o.determinizeOutput, "determinize-output", false, "Determinize dry run's output by ordering the created object and removing the unnecessary fields")
```

Comment:
This smells like "test mode" -- why is it wrong to sort fields always and why can't we ignore fields that don't matter in our call to `diff` in the tests?